pkg/build/api/validation/validation.go

```diff
@@ -225,18 +226,39 @@ func validateGitSource(git *buildapi.GitBuildSource) fielderrors.ValidationError
 	return allErrs
 }
 
-func validateImageSource(imageSource *buildapi.ImageSource) fielderrors.ValidationErrorList {
+func validateImageSources(imageSources []buildapi.ImageSource) fielderrors.ValidationErrorList {
 	allErrs := fielderrors.ValidationErrorList{}
-	allErrs = append(allErrs, validateFromImageReference(&imageSource.From).Prefix("from")...)
-	if imageSource.PullSecret != nil {
-		allErrs = append(allErrs, validateSecretRef(imageSource.PullSecret).Prefix("pullSecret")...)
-	}
-	if len(imageSource.Paths) == 0 {
-		allErrs = append(allErrs, fielderrors.NewFieldRequired("paths"))
+	for _, imageSource := range imageSources {
+		allErrs = append(allErrs, validateFromImageReference(&imageSource.From).Prefix("from")...)
```

Comment:
the prefix doesn't indicate which ImageSource in the list the error is coming from, use IndexPrefix(i) before .Prefix("from")